src/stdio_mgr/stdio_mgr.py

```diff
@@ -191,6 +205,13 @@ def stdio_mgr(in_str=""):
     sys.stdout = old_stdout
     sys.stderr = old_stderr
 
-    new_stdin.close()
+    try:
+        closed = new_stdin.closed
+    except ValueError:
```

Comment:
In what situation(s) could this raise `ValueError`?

If I read [this](https://docs.python.org/3/library/io.html#io.IOBase.close) right, `.closed` should always be a simple bool.

If `new_stdin` were undefined, it'd be a `NameError` -- if `new_stdin` were somehow the wrong type, not possessing a `closed` attribute, then it would be an `AttributeError`.

Reply:
It raises ValueError when it is detached.  Will add a note.  It is odd.

Reply:
Indeed -- it looks like touching ***most*** of the `TextIOBase` attributes, as well as those inherited from `IOBase`, raises `ValueError` after the buffer has been detached. This probably warrants augmentation of the docs; [bpo ticket created](https://bugs.python.org/issue37699).